dropgz/pkg/embed/payload.go

```diff
@@ -0,0 +1,83 @@
+package embed
+
+import (
+	"bufio"
+	"compress/gzip"
+	"embed"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+const (
+	cwd = "gz"
+)
+
+var ErrArgsMismatched = errors.New("mismatched argument count")
+
+// gzfs contains the gzipped files for deployment, as a read-only FileSystem containing only "gzfs/".
+//nolint:typecheck // dir is populated at build.
+//go:embed gz
+var gzfs embed.FS
+
+func Contents() ([]string, error) {
+	contents := []string{}
+	if err := fs.WalkDir(gzfs, cwd, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() {
+			return nil
+		}
+		contents = append(contents, strings.TrimPrefix(path, cwd+string(filepath.Separator)))
```

Comment:
nit: `cwd+string(filepath.Separator)` could be calculated once and reused for every path?